django/db/models/fields/related.py

```diff
@@ -777,14 +779,34 @@ def get_attname_column(self):
         return attname, None
 
     def get_joining_columns(self, reverse_join=False):
-        source = self.reverse_related_fields if reverse_join else self.related_fields
+        warnings.warn(
+            "The usage of get_joining_columns is deprecated. Use get_joining_fields "
+            "instead.",
+            RemovedInDjango60Warning,
+        )
         return tuple(
-            (lhs_field.column, rhs_field.column) for lhs_field, rhs_field in source
+            (lhs_field.column, rhs_field.column)
+            for lhs_field, rhs_field in self.get_joining_fields(
```

Comment:
`get_joining_columns()` cannot use `get_joining_fields()` as it will crash during the deprecation period when `get_joining_fields()` is not implemented.